engine/access/ingestion/engine.go

```diff
@@ -340,33 +382,11 @@ func (e *Engine) process(originID flow.Identifier, event interface{}) error {
 		err := e.messageHandler.Process(originID, event)
 		e.executionReceiptsNotifier.Notify()
 		return err
-	case *model.Block:
-		err := e.messageHandler.Process(originID, event)
-		e.finalizedBlockNotifier.Notify()
-		return err
 	default:
 		return fmt.Errorf("invalid event type (%T)", event)
 	}
 }
 
-// SubmitLocal submits an event originating on the local node.
-func (e *Engine) SubmitLocal(event interface{}) {
-	err := e.process(e.me.NodeID(), event)
-	if err != nil {
-		engine.LogError(e.log, err)
-	}
-}
-
-// Submit submits the given event from the node with the given origin ID
-// for processing in a non-blocking manner. It returns instantly and logs
-// a potential processing error internally when done.
-func (e *Engine) Submit(_ channels.Channel, originID flow.Identifier, event interface{}) {
-	err := e.process(originID, event)
-	if err != nil {
-		engine.LogError(e.log, err)
-	}
-}
-
 // ProcessLocal processes an event originating on the local node.
 func (e *Engine) ProcessLocal(event interface{}) error {
```

Comment:
Let's remove this one too since it's no longer used, then the engine implements `MessageProcessor` which is the modern interface
https://github.com/onflow/flow-go/blob/1ba41690954621d15d59dcabcc8f0dfc0b56f1bd/network/engine.go#L42-L59

please also add a compiler check at the top to ensure `Engine` implements it

Reply:
[abb7c00](https://github.com/onflow/flow-go/pull/5714/commits/abb7c00caa5d900eb5dd98b2e9fe6865a17af99f)